x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -0,0 +1,298 @@
+          tasks.push(async () => {
+            await jobSavedObjectService.createAnomalyDetectionJob(jobId);
+            if (datafeedId !== undefined && datafeedId !== null) {
+              // add datafeed id after saved object has been created
+              await jobSavedObjectService.addDatafeed(datafeedId, jobId);
+            }
+          });
+        }
+      }
+    }
+    for (const job of status.jobs['data-frame-analytics']) {
+      if (job.checks.savedObjectExits === false) {
+        results.savedObjectsCreated.push(job.jobId);
+        if (simulate === false) {
+          // create DFA saved objects for jobs which are missing them
+          const jobId = job.jobId;
+          tasks.push(async () => await jobSavedObjectService.createDataFrameAnalyticsJob(jobId));
+        }
+      }
+    }
+
+    for (const job of status.savedObjects['anomaly-detector']) {
+      if (job.checks.jobExists === false) {
+        results.savedObjectsDeleted.push(job.jobId);
+        if (simulate === false) {
+          // Delete AD saved objects for jobs which no longer exist
+          const jobId = job.jobId;
+          tasks.push(async () => await jobSavedObjectService.deleteAnomalyDetectionJob(jobId));
+        }
+      }
+    }
+    for (const job of status.savedObjects['data-frame-analytics']) {
+      if (job.checks.jobExists === false) {
+        results.savedObjectsDeleted.push(job.jobId);
+        if (simulate === false) {
+          // Delete DFA saved objects for jobs which no longer exist
+          const jobId = job.jobId;
+          tasks.push(async () => await jobSavedObjectService.deleteDataFrameAnalyticsJob(jobId));
+        }
+      }
+    }
+
+    for (const job of status.savedObjects['anomaly-detector']) {
+      if (job.checks.datafeedExists === true && job.datafeedId === null) {
+        // add datafeed id for jobs where the datafeed exists but the id is missing from the saved object
+        results.datafeedsAdded.push(job.jobId);
+        if (simulate === false) {
+          const df = datafeeds.datafeeds.find((d) => d.job_id === job.jobId);
+          const jobId = job.jobId;
+          const datafeedId = df?.datafeed_id;
+
+          tasks.push(async () => {
+            if (datafeedId !== undefined) {
+              await jobSavedObjectService.addDatafeed(datafeedId, jobId);
+            }
+          });
+        }
+      } else if (
+        job.checks.jobExists === true &&
+        job.checks.datafeedExists === false &&
+        job.datafeedId !== null &&
+        job.datafeedId !== undefined
+      ) {
+        // remove datafeed id for jobs where the datafeed no longer exists but the id is populated in the saved object
+        results.datafeedsRemoved.push(job.jobId);
+        if (simulate === false) {
+          const datafeedId = job.datafeedId;
+          tasks.push(async () => await jobSavedObjectService.deleteDatafeed(datafeedId));
+        }
+      }
+    }
+    await Promise.allSettled(tasks.map((t) => t()));
```

Comment:
quesiton: shall we provide a response for the failed tasks?

Reply:
good point, errors added to response in https://github.com/elastic/kibana/pull/77916/commits/6c5756fd5c023f44cc0e36b48f5986dd94dba23f